### PR TITLE
[Merged by Bors] - refactor: Make sure concrete Heyting algebras define their `himp` explicitly

### PR DESCRIPTION
**Pull request by YaelDillies**
There are a few structures in Mathlib which mathematically are Heyting algebras but do not actually define their Heyting implication, or define it implicitly as `x ⇨ y := y ⊔ xᶜ` when a nicer formula is available. This PR makes sure this is not the case.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Finsubgraph.lean [hunk 1]
```diff
@@ -59,21 +61,45 @@ instance : Sup G.Finsubgraph :=
...
-instance [Finite V] : Top G.Finsubgraph :=
-  ⟨⟨⊤, finite_univ⟩⟩
+section Finite
+variable [Finite V]
+
+instance instTop : Top G.Finsubgraph where top := ⟨⊤, finite_univ⟩
+instance instSupSet : SupSet G.Finsubgraph where sSup s := ⟨⨆ G ∈ s, ↑G, Set.toFinite _⟩
+instance instInfSet : InfSet G.Finsubgraph where sInf s := ⟨⨅ G ∈ s, ↑G, Set.toFinite _⟩
+
+@[simp, norm_cast] lemma coe_top : (⊤ : G.Finsubgraph) = (⊤ : G.Subgraph) := rfl
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/SimpleGraph/Finsubgraph.lean:82]: I'd be tempted to write the `\u`s explicitly here, just to make the naming easy to justify.

File: Mathlib/Combinatorics/SimpleGraph/Subgraph.lean [hunk 2]
```diff
@@ -879,10 +882,6 @@ taking the portion of `G` that intersects `G'`. -/
 protected abbrev restrict {G' : G.Subgraph} : G.Subgraph → G'.coe.Subgraph :=
   Subgraph.comap G'.hom
 
-@[simp]
-lemma verts_coeSubgraph {G' : Subgraph G} (G'' : Subgraph G'.coe) :
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:883]: Where did this go?

Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:883]: The prophecy...

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**